github/commitstatus.go

```diff
@@ -121,14 +130,29 @@ func (cs CommitStatus) Add(sha, state, targetURL, description string) error {
 	if err != nil {
 		return fmt.Errorf("create http request: %w", err)
 	}
-	req.Header.Set("Authorization", "token "+cs.token)
+	if !cs.useGithubAppAuth {
+		req.Header.Set("Authorization", "token "+cs.token)
+	}
 	req.Header.Set("Accept", "application/vnd.github.v3+json")
 	req.Header.Set("Content-Type", "application/json")
 
 	// The retryable unit of work.
 	workFn := func() error {
 		// By default, there is no timeout, so the call could hang forever.
 		client := &http.Client{Timeout: time.Second * 30}
+		if cs.useGithubAppAuth {
+			// We need a transport that we can update if using GitHub App authentication
+			transport := http.DefaultTransport.(*http.Transport).Clone()
+			var ghAppInstallationTransport *ghinstallation.Transport
```

Comment:
NOTE: if I am not mistaken, as it is implemented here, this would be breaking for our GitHub Enterprise users since ghinstallation would always try to generate installation token against `https://api.github.com`